hack/update-deps.sh

```diff
@@ -29,20 +29,22 @@ set -o xtrace
 cd "$(git rev-parse --show-toplevel)"
 trap 'echo "FAILED" >&2' ERR
 
-# dep itself has a problematic testdata directory with infinite symlinks which
-# makes bazel sad: https://github.com/golang/dep/pull/1412
-# dep should probably be removing it, but it doesn't:
-# https://github.com/golang/dep/issues/1580
-rm -rf vendor/github.com/golang/dep/internal/fs/testdata
-# go-bindata does too, and is not maintained ...
-rm -rf vendor/github.com/jteeuwen/go-bindata/testdata
-# docker has a contrib dir with nothing we use in it, dep will retain the licenses
-# which includes some GPL, so we manually prune this. 
-# See https://github.com/kubernetes/steering/issues/57
-rm -rf vendor/github.com/docker/docker/contrib
-bazel run //:dep -- ensure -v "$@"
-rm -rf vendor/github.com/golang/dep/internal/fs/testdata
-rm -rf vendor/github.com/jteeuwen/go-bindata/testdata
-rm -rf vendor/github.com/docker/docker/contrib
+prune-vendor() {
+  find vendor -type f \
+    -not -iname "*.go" \
+    -not -iname "*.s" \
+    -not -iname "LICENSE*" \
+    -not -iname "AUTHORS*" \
+    -not -iname "CONTRIBUTORS*" \
+    -not -iname "LICENSE*" \
```

Comment:
my bad. I duplicated this one by accident.

Reply:
... or maybe this was a test that I failed horribly? :smiling_imp: 

Also keeping .proto, .h, .c files